File: src/main/java/org/fountainmc/api/world/block/plants/Sugarcane.java
package org.fountainmc.api.world.block.plants;

import javax.annotation.Nonnegative;

public interface Sugarcane extends GrowingPlant {

    int MAXIMUM_GROWTH = 16;

    @Override
    Sugarcane withGrowth(@Nonnegative int growth);

    @Override
    default int getMaxGrowth() {
        return MAXIMUM_GROWTH;
    }

}

File: src/main/kotlin/org/fountainmc/api/world/block/plants/Sugarcane.kt
package org.fountainmc.api.world.block.plants

import org.fountainmc.api.BlockType
import javax.annotation.Nonnegative

interface Sugarcane : GrowingPlant {
    override val blockType: BlockType<Sugarcane>

    override fun withGrowth(@Nonnegative growth: Int): Sugarcane

    override val maxGrowth: Int
        get() = MAXIMUM_GROWTH

    companion object {

        val MAXIMUM_GROWTH = 16
    }

}
